### PR TITLE
Feat/co working space

### PR DESCRIPTION
**Pull request by daveeedu**
This PR implements:
- The user interface of the co-working space screens
- Responsiveness on the user interface for accessibility on all device screen sizes

### CODE REVIEW

File: components/Co-workingSpace/map.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            }
+        };
+
+        loadMap();
+    }, [address]);
+
+    return (
+        <div className='w-full'>
+            <Head>
+                <Script src={`https://maps.googleapis.com/maps/api/js?key=${process.env.NEXT_PUBLIC_GOOGLE_MAPS_API_KEY}&libraries=places`} />
```
Comment (sammy-code98) [inline components/Co-workingSpace/map.tsx:56]: `next/script `should not be used in` next/head` component. Move `<Script />` outside of `<Head>` instead.

File: pages/co-workingSpace/details.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        <title>Afrisplash</title>
+        <meta name="description" content="Generated by create next app" />
+      </Head>
+      <GeneralLayout>
+        <div className="afri-container grid gap-8 xl:grid-cols-6 md:grid-cols-5 my-8">
+          <div className="relative md:h-[100%] h-[500px] rounded-lg bg-cover md:col-span-3 xl:col-span-4 bg-no-repeat bg-[url(../assets/co-workingSpace/Office1.png)]">
+            <div className="absolute md:pl-10 pl-4  flex flex-col justify-end rounded-lg inset-0 bg-gradient-to-t from-[#D6ECDC] via-transparent to-transparent">
+              <div className="flex justify-start mb-5 md:ml-14 ml-6">
+                <Image src={calender2} alt="chatIcon" className="mr-2" />
+                <h5 className="text-white text-center text-dark_black">Joined April 2022</h5>
```
Comment (sammy-code98) [inline pages/co-workingSpace/details.tsx:37]: `text-dark_black` applies the same CSS properties as `text-white`.

please stick to one

File: pages/co-workingSpace/details.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                  <div className="text-grey_3 xlg:text-[13px] md:text-[16px] text-[10px] text-[400] mt-2 pl-5">{hours.day} :</div>
+                  {hours.startTime && hours.endTime ? (
+                    <div className="col-span-2 flex mt-2 xlg:text-[13px] md:text-[16px] text-[10px] text-[400] justify-between px-5 text-grey_3"><span className="">{hours.startTime} </span> - <span className="">{hours.endTime}</span>
+                    </div>
+                  ) : (
+                      <div className="col-span-2 xlg:text-[13px] md:text-[16px] text-[10px] text-[400] flex mt-2 justify-start px-5 text-grey_3"><span className="">Closed</span>
+                    </div>
+                  )}
+                  <div className="">
+                    <Button
```
Comment (sammy-code98) [inline pages/co-workingSpace/details.tsx:120]: can you make the reserve now button active?
 a modal is supposed to show on the click of the button.
 here is a link to the pop up design 
 https://www.figma.com/file/VxoknphZySezGNvfr63r42/Afrisplash-Final-UI?type=design&node-id=4590-52884&mode=design&t=IEncb3KeUJdpzkIp-0

File: pages/co-workingSpace/index.tsx
Comment (sammy-code98) [inline pages/co-workingSpace/index.tsx:1]: great work here David.

I did notice that the UI for your index(/co-workingSpace) is quite different from what we have on the figma design.
please take a look at the design and make the necessary changes.

https://www.figma.com/file/VxoknphZySezGNvfr63r42/Afrisplash-Final-UI?type=design&node-id=8151-89005&mode=design&t=YyfM6uUGZqc5gDKz-0


Again, super great work man

**Review by sammy-code98 — CHANGES_REQUESTED** [review]

File: pages/co-workingSpace/index.tsx
Comment (daveeedu) [reply pages/co-workingSpace/index.tsx:1]: Thank you for the feedback Sammy,

- Emmanuel requested that I should implement the screen the way it is. we took this part of the UI  

Screenshot from 2023-08-23 14-25-26

and combined it with this part from the Figma 

Screenshot from 2023-08-23 14-28-13

Kindly sync with Emmanuel to get further clarity on this. 

Thank you

File: pages/co-workingSpace/details.tsx (hunk 3 above)
Comment (daveeedu) [reply pages/co-workingSpace/details.tsx:120]: I will implement this and make a push, it was an oversight on my part.

File: pages/co-workingSpace/index.tsx
Comment (sammy-code98) [reply pages/co-workingSpace/index.tsx:1]: Alight @daveeedu , thanks for pointing that out.

You can go ahead to make the other changes requested then.
Great works far man

File: pages/co-workingSpace/details.tsx (hunk 3 above)
Comment (daveeedu) [reply pages/co-workingSpace/details.tsx:134]: Hello Sammy, This has been implemented.

File: components/Co-workingSpace/ModalComp.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
+import React from 'react'
+import Button from "@/components/atoms/Button/Button";
+import Dialog from '@mui/material/Dialog';
```
Comment (sammy-code98) [inline components/Co-workingSpace/ModalComp.tsx:3]: Hey @daveeedu, great work here man.

I noticed that you introduced another package(MUI) to the project to complete the modal component.
At the moment we are choosing to stick to using headlessUi as our extra component library/framework.
They have an excellent modal component which you can find here [HeadlessUiModal](https://headlessui.com/react/dialog)    
Or better still you can use the modal component we have that is powered by headlessUi which you can find `@components/atoms/Modal/Modal.tsx`

**Review by sammy-code98 — CHANGES_REQUESTED** [review]

File: components/Co-workingSpace/ModalComp.tsx (hunk 4 above)
Comment (sammy-code98) [reply components/Co-workingSpace/ModalComp.tsx:3]: dave
Can you also look into the heading text here, it isn't properly aligned
Thank you

Comment (daveeedu) [reply components/Co-workingSpace/ModalComp.tsx]: Hello @sammy-code98, I have made the changes you requested. Thank you for the feedback.

**Review by sammy-code98 — APPROVED** [review]

**Review by copy-ninja1 — APPROVED** [review]

**Pull request merged**